docs_src/design/adr/devops/0007-Release-Automation.md

```diff
@@ -0,0 +1,87 @@
+# Release Automation
+
+## Status
+
+In Review for Geneva
+
+## Context
+
+EdgeX Foundry is a framework composed of microservices to ease development of IoT/Edge solutions. With the framework getting richer, project growth, the number of artefacts to be released has increased. This proposal outlines the automation of the release process for the base artifacts.
+
+## Requirements
+
+### Release Artifact Definition
+
+For the scope of Geneva release artifact types are defined as:
+
+- GitHub tags in the repositories.
+- Docker images in our Nexus repository and Docker hub.
+- Snaps in the Snapcraft store.
+
+This list is likely to expand in future releases.
+
+### General Requirements
+
+As the EdgeX Release Czar I gathered the following requirements for automating this part of the release.
+
+1. The release automation needs a manual trigger to be triggered by the EdgeX Release Czar or the Linux Foundation Release Engineers. The goal of this automation is to have a "push button" release mechanism to reduce human error in our release process.
+2. Release artifacts can come from one or more GitHub repositories at a time.
+3. GitHub repositories can have one or more release artifact types to release.
+4. GitHub repositories can have one or more artifacts of a specific type to release. (For example: The mono repository, edgex-go, has more than 20 docker images to release.)
+5. GitHub repositories may be released at different times. (For example: Application and Device service repositories can be released on a different day than the Core services in the mono repository.)
+6. Ability to track multiple release streams for the project.
+7. An audit trail history for releases.
+
+## Location
+
+The code that will manage the release automation for EdgeX Foundry will live in a repository called `cd-management`. This repository will have a branch named `release` that will track the releases of artifacts off the `master` branch of the EdgeX Foundry repositories.
+
+### Multiple Release Streams
+
+EdgeX Foundry has this idea of multple release streams that basically coincides with different named branches in GitHub. For the majority of the main releases we will be targeting those off the `master` branch. In the future we will mark a specific release for long term support (LTS). When this happens we will have to branch off `master` in the EdgeX repositories and create a separate release stream for the LTS. The suggestion at that point will be to branch off the `release` branch in `cd-management` and use this new release branch to track the LTS branches in the EdgeX repositories.
```

Comment:
This is the first mention of the release branch in cd-management, so I don't quite follow this last statement...

Reply:
The logic will should be the same for both branches. The main reason I'm proposing to branch off is to separate the releases of each branch. We have a lot of artifacts we are releasing and having two release streams in one branch will be very verbose and hard to follow. It also separates the audit logs for the releases.